### ISSUE TITLE
H5 volume data storage order is wrong

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#1141]
HDF5 uses arrays with the last dimension varying fastest, but we vary the first dimension of points within elements fastest, so when we write data to an array it ends up shaped wrong.

For example, the z component of a non-rotated 5x5x2-point brick is (from h5py)
```
[array([[ 0.,  0.], [ 0.,  0.], [ 0.,  0.], [ 0.,  0.], [ 0.,  0.]]),
 array([[ 0.,  0.], [ 0.,  0.], [ 0.,  0.], [ 0.,  0.], [ 0.,  0.]]),
 array([[ 0.,  0.], [ 0.,  0.], [ 0.,  1.], [ 1.,  1.], [ 1.,  1.]]),
 array([[ 1.,  1.], [ 1.,  1.], [ 1.,  1.], [ 1.,  1.], [ 1.,  1.]]),
 array([[ 1.,  1.], [ 1.,  1.], [ 1.,  1.], [ 1.,  1.], [ 1.,  1.]])]
```

**Comment by nilsdeppe** [comment]
Things visualize in Paraview correctly as far as I can tell. Did you find something in the HDF5 documentation that describes assumed ordering? It could be that h5py makes an assumption.

**Comment by nilsdeppe** [comment]
I observe what you see what I run `h5dump`. I'm wondering if it really makes sense for us to assign a more than rank 1 representation to the x, y, and z coords (and tensor components). The reason is we store them as contiguous x,y,z they probably should just be rank 1 datasets with the order as we defined.

Edit:

Some related links I found:
https://www.harrisgeospatial.com/docs/HDF5_Overview.html
https://support.hdfgroup.org/HDF5/doc/H5.intro.html
